handlers/filter.go

```diff
@@ -0,0 +1,73 @@
+package handlers
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"strings"
+
+	"github.com/ONSdigital/dp-frontend-search-controller/data"
+	"github.com/ONSdigital/log.go/log"
+)
+
+func mapSubFilterTypes(ctx context.Context, query url.Values) (apiQuery url.Values, err error) {
+	apiQuery, err = url.ParseQuery(query.Encode())
+	if err != nil {
+		log.Event(ctx, "failed to parse copy of query for mapping filter types", log.Error(err), log.ERROR)
+		return nil, err
+	}
+	filters := apiQuery["filter"]
+	if len(filters) > 0 {
+		var newFilters []string
```

Comment:
Dave, thank you so much for sharing this. I had no idea about this as well. But whilst reading through the link, I realised that it is better to initialise an empty slice rather than a nil slice. When thinking in a contextual way, a nil slice represents a slice that doesn’t exist but in this case, `newFilters` should be a string slice that should always exist and it should be empty. 

Even though, `newFilters` doesn't go through a `json.Marshal()` and the built-in functions behave the same way for both empty and nil slice, in a contextual aspect, empty slice makes more sense